local-interchain/bash/source.bash

```diff
@@ -0,0 +1,184 @@
+# NOTE: if using a /, make sure to escape it with \
+ICT_SH_EXEC() {
+    API=$1
+    CHAIN_ID=$2
+    CMD=$3 # can be multiple words
+    ICT_MAKE_REQUEST "$API" $CHAIN_ID "exec" "$CMD"
+}
+
+# === RELAYER ===
+
+# ICT_RELAYER_STOP http://127.0.0.1 "localjuno-1"
+ICT_RELAYER_STOP() {
+    API=$1
+    CHAIN_ID=$2
+
+    # TODO: how does this function?
+    ICT_MAKE_REQUEST $API $CHAIN_ID "stop-relayer" ""
+}
+
+# ICT_RELAYER_START http://127.0.0.1 "localjuno-1" "demo-path2 --max-tx-size 10"
+ICT_RELAYER_START() {
+    API=$1
+    CHAIN_ID=$2
+    CMD=$3
+
+    ICT_MAKE_REQUEST $API $CHAIN_ID "start-relayer" "$CMD"
+}
+
+# RELAYER_EXEC http://127.0.0.1:8080 "localjuno-1" "rly paths list"
+ICT_RELAYER_EXEC() {
+    API=$1
+    CHAIN_ID=$2
+    shift 2 # get the 3rd argument and up as the command
+    CMD="$*"
+
+    ICT_MAKE_REQUEST $API $CHAIN_ID "relayer-exec" "$CMD"
+}
+
+# RELAYER_CHANNELS http://127.0.0.1:8080 "localjuno-1"
+ICT_RELAYER_CHANNELS() {
+    API=$1
+    CHAIN_ID=$2
+
+    ICT_MAKE_REQUEST $API $CHAIN_ID "get_channels" ""
+}
+
+# === COSMWASM ===
+
+# ICT_WASM_DUMP_CONTRACT_STATE "http://localhost:8080" "localjuno-1" "cosmos1contractaddress" "100"
+ICT_WASM_DUMP_CONTRACT_STATE() {
+    API=$1
+    CHAIN_ID=$2
+    CONTRACT=$3
+    HEIGHT=$4
+
+    ICT_MAKE_REQUEST $API $CHAIN_ID "recover-key" "contract=$CONTRACT;height=$HEIGHT"
+}
+
+# ICT_WASM_STORE_FILE "http://localhost:8080" "localjuno-1" "/host/absolute/path.wasm" "keyName"
+# returns the code_id of the uploaded contract
+ICT_WASM_STORE_FILE() {
+    API=$1
+    CHAIN_ID=$2
+    FILE=$3
+    KEYNAME=$4
+
+    DATA=`printf '{"chain_id":"%s","file_path":"%s","key_name":"%s"}' $CHAIN_ID $FILE $KEYNAME`
+    curl "$API/upload" --header "Content-Type: application/json" --header "Upload-Type: cosmwasm" -X POST -d "$DATA"
+}
+
+# === OTHER ===
+
+# ICT_POLL_FOR_START "http://localhost:8080" 50
+ICT_POLL_FOR_START() {
+    API=$1
+    ATTEMPTS_MAX=$2
+
+    curl --head -X GET --retry $ATTEMPTS_MAX --retry-connrefused --retry-delay 3 $API
+}
+
+# ICT_KILL_ALL "http://localhost:8080" "localjuno-1"
+# (Kills all running, keeps local-ic process. `killall local-ic` to kill that as well)
+ICT_KILL_ALL() {
+    API=$1
+    CHAIN_ID=$2
+    ICT_MAKE_REQUEST $API $CHAIN_ID "kill-all" ""
+}
+
+# ICT_GET_PEER "http://localhost:8080" "localjuno-1"
+ICT_GET_PEER() {
+    API=$1
+    CHAIN_ID=$2
+
+    if [[ $API != */info ]]; then
+        API="$API/info"
+    fi
+
+    curl -G -d "chain_id=$CHAIN_ID" -d "request=peer" $API
+}
+
+# ICT_FAUCET_REQUEST "http://localhost:8080" "localjuno-1" "1000000000ujuno" "juno1qk7zqy3k2v3jx2zq2z2zq2zq2zq2zq2zq2zq"
+ICT_FAUCET_REQUEST() {
+    API=$1
+    CHAIN_ID=$2
+    AMOUNT=$3
+    ADDRESS=$4
+    ICT_MAKE_REQUEST $API $CHAIN_ID "faucet" "amount=$AMOUNT;address=$ADDRESS"
+}
+
+# ICT_ADD_FULL_NODE http://127.0.0.1:8080 "localjuno-1" "1"
+ICT_ADD_FULL_NODE() {
+    API=$1
+    CHAIN_ID=$2
+    AMOUNT=$3
```

Comment:
I would consider using `local`

`local API=$1, CHAIN_ID=$2, AMOUNT=$3`

in this way variables will not leak into outside context which might make debugging harder especially if multiple functions are called in succession with different args

Reply:
good call

Reply:
resolved in: 92e712f80dc0db60ce8cf058c7c0e9c5d331f58a